### ISSUE TITLE
Issues building under Ubuntu 16.04

### ISSUE THREAD

**Issue by aapo** [etodd/yearning#2]
- Readme says "./setup", it is "setup-linux"
- It doesn't have execute bit (chmod a+x setup-linux;./setup-linux)
- Tries to install "libXmu-dev", correct is "libxmu-dev"
- ("apt-get install" is very Ubuntu/Debian -style. Maybe split setup-linux really containing building on any Linux derivative + another file for dependencies)
- Version number of Blender is not tested.
- Blender is only tested dependency. Why not just list it same way than other dependencies?

Ubuntu 16.04
make fails with:

```
Building C object external/sdl/CMakeFiles/SDL2-static.dir/src/video/wayland/SDL_waylandevents.c.o
yearning/external/sdl/src/video/wayland/SDL_waylandevents.c:40:10: fatal error: 'pointer-constraints-unstable-v1-client-protocol.h' file not found
#include "pointer-constraints-unstable-v1-client-protocol.h"
```

I tested compiling external-sdl alone:

```
cd external/sdl
mkdir build
cd build
cmake ..
make
```

And it fails with the same error.

I can rebuild libsdl2-2.0.4 (from Ubuntu 16.04) without issues.

**Comment by etodd** [comment]
Hey, thanks for submitting this. I went ahead and addressed the low-hanging fruit in f9f1173. I think the SDL issue is due to their experimental Wayland support. I was building against the bleeding edge because I keep hoping they'll fix some major issues I have on Mac and Windows. I'm now building against the official 2.0.5 release, so hopefully that's stable enough.

I'm running into another issue with Wwise right now, not sure what that's about. I'm not able to keep the Linux build as up-to-date as I'd like because the drivers for my AMD card are horrible and I'm stuck with a VM for now. The headless server build (`make yearningsrv`) should work fine.

Note that everything is extremely WIPpy. There are massive changes currently in progress. :)

**Comment by aapo** [comment]
Yes, _make yearningsrv_ works.
- One more prerequirements (for downloading the code):
  https://packagecloud.io/github/git-lfs/install
  (without it files external/wwise/_/Debug/_ are not cloned)
- submodule "src/platform/ps" points to non-existing(?) bitbucket-repo
- (At this moment) I skipped part _3. Copy the soundbanks into assets/audio/GeneratedSoundBanks/Linux_ using just  _mkdir -p assets/audio/GeneratedSoundBanks/Linux_ (Will this cause problems?)
- Now make fails:
  
  Linking CXX executable yearning
  `../external/wwise/Linux/Debug/libCommunicationCentral.a(Communication.o): In function 'AK::Comm::Init(AkCommSettings const&)':
  /var/jenkins/ws/wwise_v2016.1/Wwise/SDK/source/SoundEngine/Communication/Remote/Linux/../CodeBase/Communication.cpp:134: undefined reference to ´AkCreateProxyFramework(int)'
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  CMakeFiles/yearning.dir/build.make:1214: recipe for target 'yearning' failed`

**Comment by etodd** [comment]
- Ignore the "ps" submodule. That's PS4-specific code that I'm too scared to open-source due to NDAs.
- It will build without the soundbanks, but I think it will crash at the moment. Haven't tested it yet.
- Your link error is the same one I'm getting. Still trying to figure it out. It's related to an internal Wwise function so I'm not sure what's going on yet.
- I updated the readme with info about Git LFS and submodules. Thanks.

**Comment by etodd** [comment]
Finally fixed this via 163125e
